doc/source/whatsnew/v1.6.0.rst

```diff
@@ -116,6 +116,7 @@ See :ref:`install.dependencies` and :ref:`install.optional_dependencies` for mor
 
 Other API changes
 ^^^^^^^^^^^^^^^^^
+- Disallow passing non-round floats to :class:`Timestamp` with ``unit="M"`` or ``unit="Y"`` (:issue:`47266`)
```

Comment:
Could you start a new section in the whatsnew for these like we did in 1.0? https://pandas.pydata.org/docs/whatsnew/v1.0.0.html#removal-of-prior-version-deprecations-changes

Reply:
updated + green

Reply:
after this is merged ill update the other deprecation-enforcing PRs to move the notes to the appropriate section